### DISCUSSION TITLE
apply_access_property implementation doubt

### DISCUSSION THREAD

**Discussion by sicario001** [NVIDIA/cccl#1125] [libcu++] [answered]
The following [code snippet](https://github.com/NVIDIA/cccl/blob/61c328aee025a43045ed2f971ec3ec3fbd6bf695/libcudacxx/include/cuda/annotated_ptr#L150C1-L152C1) is part of the apply_access_property implementation. My concern is that doesn't this apply the evict_last property to addresses in multiples of **_LINE_SIZE*_LINE_SIZE = 128*128**? Shouldn't we actually do this in multiples of **_LINE_SIZE**? That would require replacing `__i += _LINE_SIZE` with `__i += 1`.
Please correct me if I am wrong if I am missing something.
Thanks

``` 
    for (std::size_t __i = 0; __i < __end; __i += _LINE_SIZE) {
      asm volatile ("prefetch.global.L2::evict_last [%0];" ::"l"(__p + (__i * _LINE_SIZE)) :);
    }
```

**Comment by wmaxey** [comment]
At first glance, this does seem like a bug. Converting to issue.

**Reply by wmaxey** [reply, marked as answer]
Nearly a year later, I have actually converted this to an issue and have opened a PR to fix the erroneous stride: #2605
